File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.swift
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        guard !nums.isEmpty else { return 0 }
        
        var pointer = 0
        for i in 1..<nums.count {
            if nums[pointer] != nums[i] {
                pointer += 1
                nums[pointer] = nums[i]
            }
        }
        return pointer + 1
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    ln := len(nums)
    if ln <= 1 {
        return ln
    }
    j := 0
    for i := 1; i < ln; i++ {
        if nums[j] != nums[i] {
            j++
            nums[j] = nums[i]
        }
    }
    return j + 1
}